test-utils/src/array_gen/binary.rs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use arrow::array::{
+    ArrayRef, BinaryViewArray, GenericBinaryArray, OffsetSizeTrait, UInt32Array,
+};
+use arrow::compute;
+use rand::rngs::StdRng;
+use rand::Rng;
+
+/// Randomly generate binary arrays
+pub struct BinaryArrayGenerator {
+    /// The maximum length of each binary element
+    pub max_len: usize,
+    /// The total number of binaries in the output
+    pub num_binaries: usize,
+    /// The number of distinct binary values in the columns
+    pub num_distinct_binaries: usize,
+    /// The percentage of nulls in the columns
+    pub null_pct: f64,
+    /// Random number generator
+    pub rng: StdRng,
+}
+
+impl BinaryArrayGenerator {
```

Comment:
Love it, thinking of if we should tests for this generator?

Reply:
the generator itself is part of a test 🤔  What would we test? Maybe that the distinct values are as specified?